internal/testutils/testutils_helpers/terraform_plan_helpers.go

```diff
@@ -0,0 +1,151 @@
+package testutils_helpers
+
+import (
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	"github.com/pingidentity/pingctl/internal/connector"
+	"github.com/pingidentity/pingctl/internal/connector/pingone/sso"
+)
+
+// Helper function to run terraform plan --generate-config-out on a single resource
+func TestSingleResourceTerraformPlanGenerateConfigOut(t *testing.T, resource connector.ExportableResource, sdkClientInfo *connector.SDKClientInfo) {
+	t.Helper()
+
+	// Get an instance of the PingOne SSO Connector
+	ssoConnector := sso.SSOConnector(sdkClientInfo.Context, sdkClientInfo.ApiClient, sdkClientInfo.ApiClientId, sdkClientInfo.ExportEnvironmentID)
+
+	// Create temporary directories for export files and terraform plan testing
+	exportDir := createTempExportDir(t, resource.ResourceType())
+
+	// Check if terraform is installed
+	terraformExecutableFilepath := checkTerraformInstallPath(t)
+
+	// Terraform Initialize the testing directory for terraform plan testing
+	initTerraformInDir(t, exportDir, terraformExecutableFilepath)
+
+	// Export the resource
+	if err := ssoConnector.ExportSingle(connector.ENUMEXPORTFORMAT_HCL, exportDir, true, resource); err != nil {
```

Comment:
The connector type is independent of resources when exporting single resources. Should we remove the `ExportSingle()` functions from the connectors, and move the logic into this file as a helper function directly, since the CLI does not need to export individual resources yet?

Reply:
It was resolved by using `WriteFiles()` directly.